src/main/webapp/app/entities/flair-bi/rightnavbar/flair-bi-rightnavbar.controller.js

```diff
@@ -318,6 +332,33 @@
             $scope.$on("$destroy", unsubscribe);
         }
 
+        function registerAddFilterInFinterPenal() {
+            var unsubscribe = $scope.$on(
+                "flairbiApp:add-filter-In-FinterPenal",
+                function () {
+                    var filterParameters;
+                    filterParameters = filterParametersService.get();
+                    var filter = Object.keys(filterParameters);
+
+                    filter.forEach(element => {
+                        vm.dimensions.map(function (item) {
```

Comment:
do we need to loops...can you do this using single loop foreach (key.val) ot filter function